www/js/Controller/My_item/Myitems.js

```diff
@@ -128,7 +128,7 @@ elephant.controller('MyitemsController', function($scope, $http, $timeout, $loca
           $scope.reApprove(dataString);
           hideSheet();
         }else if(index == 1){
-          $scope.deleteItem(dataString, item);
+          $scope.deleteGivenAway(dataString, item);
```

Comment:
it looks like deleteGivenAway(x,y) is a private method as the user is not accessing it directly and its reApprove() function which is triggering it. In this case it's good to not have methods like this linked to $scope. As $scope is loaded when the class is initialised, the class loads all the $scope methods even if they are not directly required by scope. So u can change it to var deleteGivenAway =  function {}